pkg/api/playlist.go

```diff
@@ -2,15 +2,74 @@ package api
 
 import (
 	"net/http"
+	"strings"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
 
 	"github.com/grafana/grafana/pkg/api/dtos"
 	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/apis/playlist/v0alpha1"
+	"github.com/grafana/grafana/pkg/middleware"
 	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/grafana-apiserver/endpoints/request"
 	"github.com/grafana/grafana/pkg/services/playlist"
 	"github.com/grafana/grafana/pkg/web"
 )
 
-func (hs *HTTPServer) ValidateOrgPlaylist(c *contextmodel.ReqContext) {
+func (hs *HTTPServer) registerPlaylistAPI(apiRoute routing.RouteRegister) {
+	searchHandler := routing.Wrap(hs.SearchPlaylists)
+	if hs.Features.IsEnabled(featuremgmt.FlagKubernetesPlaylistsAPI) {
+		namespacer := request.GetNamespaceMapper(hs.Cfg)
+		gvr := schema.GroupVersionResource{
+			Group:    v0alpha1.GroupName,
+			Version:  v0alpha1.VersionID,
+			Resource: "playlists",
+		}
+
+		// For now -- test a single endpoint
+		searchHandler = func(c *contextmodel.ReqContext) response.Response {
+			dyn, err := dynamic.NewForConfig(hs.clientConfigProvider.GetDirectRestConfig(c))
+			if err != nil {
+				return response.Error(http.StatusInternalServerError, "config", err)
+			}
+			client := dyn.Resource(gvr).Namespace(namespacer(c.OrgID))
+			out, err := client.List(c.Req.Context(), v1.ListOptions{})
+			if err != nil {
+				return response.Error(http.StatusBadRequest, "list error", err)
+			}
+
+			query := strings.ToUpper(c.Query("query"))
+			playlists := []playlist.Playlist{}
+			for _, item := range out.Items {
+				p := v0alpha1.UnstructuredToLegacyPlaylist(&item)
+				if p == nil {
+					continue
+				}
+				if query != "" && !strings.Contains(strings.ToUpper(p.Name), query) {
+					continue // query filter
+				}
+				playlists = append(playlists, *p)
+			}
+			return response.JSON(http.StatusOK, playlists)
+		}
+	}
+
+	// Playlist API
+	apiRoute.Group("/playlists", func(playlistRoute routing.RouteRegister) {
```

Comment:
After https://github.com/grafana/grafana/pull/75709 -- we can replace all endpoints